cmd/ctr/commands/run/run_unix.go

```diff
@@ -115,12 +124,42 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 			opts = append(opts, oci.WithImageConfig(image))
 			cOpts = append(cOpts,
 				containerd.WithImage(image),
-				containerd.WithSnapshotter(snapshotter),
+				containerd.WithSnapshotter(snapshotter))
+			if uidmap, gidmap := context.String("uidmap"), context.String("gidmap"); uidmap != "" && gidmap != "" {
+				cUID, hUID, uSize, err := parseRemapping(uidmap, "uidmap")
+				if err != nil {
+					return nil, err
+				}
+				cGID, hGID, gSize, err := parseRemapping(gidmap, "gidmap")
+				if err != nil {
+					return nil, err
+				}
+				opts = append(opts,
+					oci.WithUserNamespace([]specs.LinuxIDMapping{
+						{
+							ContainerID: cUID,
+							HostID:      hUID,
+							Size:        uSize,
+						},
+					}, []specs.LinuxIDMapping{
+						{
+							ContainerID: cGID,
+							HostID:      hGID,
+							Size:        gSize,
+						},
+					}))
+				if context.Bool("read-only") {
+					cOpts = append(cOpts, containerd.WithRemappedSnapshotView(id, image, hUID, hGID))
```

Comment:
if you return a struct from the `parseRemapping` you can just use the `HostID` values directly from the struct for these two `...SnapshotView()` functions.

Reply:
:+1: